sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultCredentialPolicyTest.java

```diff
@@ -381,6 +402,160 @@ public void onAuthorizeRequestDifferentScopeVerifyFalse() {
         assertTrue(onChallenge);
     }
 
+    // Normal flow: 401 Unauthorized -> 200 OK -> 401 Unauthorized with claims -> 200 OK
+    @SyncAsyncTest
+    public void processMultipleResponses() {
+        MutableTestCredential testCredential = new MutableTestCredential(new ArrayList<>(BASE_ASSERTIONS))
+            .addAssertion(tokenRequestContext -> tokenRequestContext.getClaims() == null);
+        HttpResponse[] responses = new HttpResponse[] {
+            unauthorizedHttpResponseWithHeader,
+            simpleResponse,
+            unauthorizedHttpResponseWithHeaderAndClaims,
+            simpleResponse
+        };
+        AtomicInteger currentResponse = new AtomicInteger();
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(testCredential, false);
+
+        HttpPipeline pipeline = new HttpPipelineBuilder()
+            .policies(policy)
+            .httpClient(ignored -> Mono.just(responses[currentResponse.getAndIncrement()]))
+            .build();
+
+        // The first request to a Key Vault endpoint without an access token will always return a 401 Unauthorized
+        // response with a WWW-Authenticate header containing an authentication challenge.
+
+        HttpResponse firstResponse = SyncAsyncExtension.execute(
+            () -> pipeline.sendSync(this.callContext.getHttpRequest(), this.callContext.getContext()),
+            () -> pipeline.send(this.callContext.getHttpRequest(), this.callContext.getContext()));
+
+        String firstToken = this.callContext.getHttpRequest().getHeaders().getValue(AUTHORIZATION);
+
+        // The first response was unauthorized and a token was set on the request.
+        assertNotNull(firstToken);
+        // On a second attempt, a successful response was received.
+        assertEquals(simpleResponse, firstResponse);
+
+        testCredential.replaceAssertion(tokenRequestContext -> tokenRequestContext.getClaims() != null, 3);
```

Comment:
Can you assert that the actual value is correct i.e. `{"access_token":{"acrs":{"essential":true,"value":"cp1"}}}`?